### DISCUSSION TITLE
DataSet.Serialize Fields with JSON Values

### DISCUSSION THREAD

**Discussion by idrall-lgjaramillo** [viniciussanchez/dataset-serialize#201] [Ideas]
Olá Vinicius, gostaria de comentar com você que é conveniente para mim adicionar um tratamento para retornar os valores dos campos JSON como Objetos e não como strings.

No meu caso estou criando microsserviços e é comum o banco de dados manipular campos JSON então tive que adicionar essas linhas na unidade DataSet.Serialize.Export.pas (da linha 303)

```
        begin
        LJSonValue := TJSonObject.ParseJSONValue(LField.AsWideString);
        if ( LJSonValue <> nil) then
          Result.{$IF DEFINED(FPC)}Add{$ELSE}AddPair{$ENDIF}(LKey, LJSonValue)
        else
          Result.{$IF DEFINED(FPC)}Add{$ELSE}AddPair{$ENDIF}(LKey, TJSONString.Create(LField.AsWideString));
        end;
```

**Comment by viniciussanchez** [comment]
@idrall-lgjaramillo bom dia... poderia verificar se seu código está atualizado por favor? Porque eu procurei aqui por essa linha 303 e está bem diferente do código que você colocou acima. Qualquer coisa atualiza o seu código, e verifica se vai resolver. Caso contrário, pode até enviar seu pull request, ou então colocar o código com mais detalhes aqui por favor

![image](https://github.com/viniciussanchez/dataset-serialize/assets/16382981/8f870e63-5490-470b-8d4a-a3cfb1bebe07)
